features/steps/shopcart_steps.py

```diff
@@ -13,3 +13,13 @@ def step_impl(context, message):
 @then(u'I should not see "{message}"')
 def step_impl(context, message):
     assert message not in context.resp.data
+
+@when(u'I visit "{url}"')
+def step_impl(context, url):
+    context.resp = context.app.get(url)
+    assert context.resp.status_code == 200
+
+@when(u'I search shopcarts with uid "{uid}"')
+def step_impl(context, uid):
```

Comment:
Also I think 'when' should just do get, post, delete etc. and whatever happens from it should be in 'then'